### ISSUE TITLE
Generalise storage and collaboration backends

### ISSUE THREAD

**Issue by MarZab** [excalidraw/excalidraw#6426]
I would like to propose the excalidraw app gets a abstraction layer for firebase and socket.io.

While this will also provide a good abstraction for testing and future development, my primary intent is to implement alternative backends for storing sessions/files/scenes and for real-time collaboration. 

My proposal is to refactor all mentions of firebase and socket.io into a class type and expose it via a property, that defaults to the current behaviour to preserve existing features.

A good start was made here for the storage part: https://github.com/alswl/excalidraw/pull/2/files

References:
- https://github.com/excalidraw/excalidraw/issues/1772
- https://github.com/excalidraw/excalidraw/issues/4038

**Comment by ddompe** [comment]
What storage backend do you use for this http storage you are deploying @MarZab ?

**Comment by MarZab** [comment]
I would personally like to use AWS Api Gateway Websockets @ddompe

**Comment by vpzomtrrfrt** [comment]
Is there a reason that couldn't be merged? Or this [this comment](https://github.com/excalidraw/excalidraw/issues/1772#issuecomment-974086405) still the official stance?

**Comment by PatWie** [comment]
@MarZab  DynamoDB (a great DB in general) has a limit of each item being constrained to max 400KB. Are you sure this is sufficient?

**Comment by MarZab** [comment]
Indeed, you can't store the whole drawing inside one record, my plan was to offload to S3 in an attempt to keep the whole backend serverless.

BTW good work on the Go server :)

**Comment by alswl** [comment]
@MarZab So cool project, go-complete.
Is it possible provide simple Go HTTP server for storage and socket collaboration(AKA excalidraw-server)? I'm interested in it.
